### PR TITLE
Feature/aut 1644/improve ux with invalid identifiers int test

### PR DESCRIPTION
**Pull request by Silvia-loza**
Related to Related to
https://oat-sa.atlassian.net/browse/AUT-1644

Improve user experience when invalid identifiers found on test settings for:

**Ac's 1 to 4**

when mandatory fields are invalid the behavior should be as per mockup: 
[Mockup for AC1 AC2 AC3 and AC4](https://invis.io/YM128I99BEKD#/462949865_01)

This behavior **implemented for:**
Test properties
Test Part:
section
subsections
Item 

**AC'S 5 & 6** 
Manage items and preview should be active even when invalid identifiers however should show a different pop up  depending on the case, see mock up for reference

https://projects.invisionapp.com/share/YM128I99BEKD#/screens/462950206

image

EXPECTED RESULT:
All to follow guidelines described in ticket and mock-up

### CODE REVIEW

File: views/js/controller/creator/views/property.js [hunk 1]
```diff
@@ -139,17 +139,58 @@ define(['jquery', 'uikitLoader', 'core/databinder', 'taoQtiTest/controller/creat
          * @private
          */
         function propValidation() {
-            $view.on('validated.group', function (e, isValid) {
-                let $testSection = $('.tlb-button-on').parents('.section').attr('id');
-                let testSectionId = `#${$testSection}`;
-                let $propsSectionError = $(`#section-props-${$testSection}`).find('span.validate-error');
+            $view.on('validated.group', function(e, isValid){
+                const activeButton = $('.tlb-button-on');
```
Comment (bziondik) [inline views/js/controller/creator/views/property.js:143]: const errors = $(e.currentTarget).find('span.validate-error');
const idOfElement = ($(e.currentTarget).find('span[data-bind="identifier"]') - cat "props-" part)

File: views/js/controller/creator/creator.js [hunk 2]
```diff
@@ -222,7 +225,7 @@ define([
                     });
 
                     creatorContext.on('preview', function() {
-                        if(isTestContainsItems()) {
+                        if(isTestContainsItems() && !isTestHasErrors()) {
```
Comment (bziondik) [inline views/js/controller/creator/creator.js:228]: use `creatorContext.isTestHasErrors()`

File: views/js/controller/creator/views/property.js [hunk 3]
```diff
@@ -139,17 +139,27 @@ define(['jquery', 'uikitLoader', 'core/databinder', 'taoQtiTest/controller/creat
...
-                let $testSection = $('.tlb-button-on').parents('.section').attr('id');
-                let testSectionId = `#${$testSection}`;
-                let $propsSectionError = $(`#section-props-${$testSection}`).find('span.validate-error');
-                let $propsItemError = $('.itemref-props').find('span.validate-error');
-
-                if (e.namespace === 'group') {
-                    if (isValid && $propsItemError.length === 0 && $propsSectionError.length === 0) {
-                        $(testSectionId).removeClass('section-error');
+            $view.on('validated.group', function(e, isValid){
+                const $warningIcon = $('span.icon-warning');
```
Comment (bziondik) [inline views/js/controller/creator/views/property.js:143]:
```suggestion
                const warningIconSelector = 'span.icon-warning';
```

File: views/js/controller/creator/views/property.js [hunk 4]
```diff
@@ -139,17 +139,27 @@ define(['jquery', 'uikitLoader', 'core/databinder', 'taoQtiTest/controller/creat
...
+
+                // finds error current element if any
+                let errors = $(e.currentTarget).find('span.validate-error');
+                let currentTargetId = '#' + $(e.currentTarget).find('span[data-bind="identifier"]').attr('id').slice(6);
+                
+                if(e.namespace === 'group'){
+                    if (isValid && errors.length === 0) {
+                        //remove warning icon if validation fails
+                        if($(e.currentTarget).hasClass('test-props')){
+                         $($test).find($warningIcon).first().css('display', 'none');
```
Comment (bziondik) [inline views/js/controller/creator/views/property.js:154]:
```suggestion
                         $test.find(warningIconSelector).first().css('display', 'none');
```

File: views/js/controller/creator/views/property.js [hunk 5]
```diff
@@ -139,17 +139,27 @@ define(['jquery', 'uikitLoader', 'core/databinder', 'taoQtiTest/controller/creat
...
+                        }
+                        $(currentTargetId).find($warningIcon).first().css('display', 'none');
                     } else {
-                        $(testSectionId).addClass('section-error');
+                       //add warning icon if validation fails
+                        if($(e.currentTarget).hasClass('test-props')){
+                            $($test).find($warningIcon).first().css('display', 'inline');
```
Comment (bziondik) [inline views/js/controller/creator/views/property.js:160]:
```suggestion
                            $test.find(warningIconSelector ).first().css('display', 'inline');
```

File: views/js/controller/creator/views/property.js [hunk 6]
```diff
@@ -139,17 +139,27 @@ define(['jquery', 'uikitLoader', 'core/databinder', 'taoQtiTest/controller/creat
...
-                if (e.namespace === 'group') {
-                    if (isValid && $propsItemError.length === 0 && $propsSectionError.length === 0) {
-                        $(testSectionId).removeClass('section-error');
+            $view.on('validated.group', function(e, isValid){
+                const $warningIcon = $('span.icon-warning');
+                const $test = $('.tlb-button-on').parents('.test-creator-test');
+
+                // finds error current element if any
+                let errors = $(e.currentTarget).find('span.validate-error');
+                let currentTargetId = '#' + $(e.currentTarget).find('span[data-bind="identifier"]').attr('id').slice(6);
```
Comment (bziondik) [inline views/js/controller/creator/views/property.js:148]:
```suggestion
                let currentTargetId = '[id="' + $(e.currentTarget).find('span[data-bind="identifier"]').attr('id').slice(6) + '"]';
```
or better use template strings
and jquery has issue to select id with dot , example '#ab.cd', should be used [id="ab.cd"]
and now in id allowed dots

**Review by bziondik — APPROVED** [review]
- [ ]  New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [x]  Feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request title and description are meaningful

**Review by clagosarias — APPROVED** [review]
Awesome job! 💪🏻

- [ ] New code is covered by tests (if applicable)
- [ ]  Tests are running successfully (old and new ones) on my local machine (if applicable)
- [x]  New code is respecting code style rules
- [x]  New code is respecting best practices
- [x]  New code is not subject to concurrency issues (if applicable)
- [ ]  The feature is working correctly on my local machine (if applicable)
- [x]  Acceptance criteria are respected
- [x]  Pull request's target is not master

**Pull request merged**